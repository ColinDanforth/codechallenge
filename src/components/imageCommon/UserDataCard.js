import React from 'react'
import PropTypes from 'prop-types'
import like from '../../localImages/like.png'

const rootDataCardStyle={
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'flex-start',
  maxWidth: '820px',
  minWidth: '820px',
  height: 'auto',
  backgroundColor: '#FFF',
  borderBottom: 'inset',
  borderRight: 'inset',
}

const textBlock = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  maxWidth: '700px',
  minWidth: '700px',
  marginLeft: '10px',
  width: 'auto',
  height: 'auto',
}

const imageHeight = {
  maxHeight: '100px',
  width: 'auto',
  height: 'auto',
  margin: '0.3%',
}

const pStyle = {
  maxWidth: '400px',
  minWidth: '400px',
  width: 'auto',
  height: 'auto',
}

const likeStyle = {
  maxWidth: '40px',
  width: 'auto',
  height: 'auto',
  marginTop: '5px',
  marginRight: '10px',
}

const UserDataCard = (props) =>{
  if(props.dataObject.hasOwnProperty('user')) {
    return (
      <div style={rootDataCardStyle}>
        <img style={imageHeight} src={props.dataObject.user.avatars.default.https} alt="avatar"/>
        <div style={textBlock}>
          <div>
            <p style={pStyle}>Photographer: {props.dataObject.user.fullname}</p>
            <p style={pStyle}>Rating: {props.dataObject.rating}%</p>
          </div>
          <div>
            <img style={likeStyle} src={like} alt='like'/>
          </div>
        </div>
      </div>
    )
  }else{
    return <div style={{'display': 'none'}}/>
  }
}

export default UserDataCard

UserDataCard.propTypes={
  dataObject: PropTypes.object
}
