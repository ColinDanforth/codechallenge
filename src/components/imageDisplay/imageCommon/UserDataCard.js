import React from 'react'
import PropTypes from 'prop-types'
import like from '../../localImges/like.png'

const rootDataCardStyle={
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'flex-start',
  height: 'auto',
  backgroundColor: '#FFF',
  borderTop: 'inset',
  borderLeft: 'inset',
  borderBottom: 'inset'
}

const textBlock = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  maxWidth: '700px',
  minWidth: '700px',
  marginLeft: '10px',
  width: 'auto',
  height: 'auto',
}

const imageHeight = {
  maxHeight: '100px',
  width: 'auto',
  height: 'auto',
  margin: '0.3%',
}

const pStyle = {
  maxWidth: '400px',
  minWidth: '400px',
  width: 'auto',
  height: 'auto',
}

const likeStyle = {
  maxWidth: '40px',
  width: 'auto',
  height: 'auto',
  marginTop: '5px',
  marginRight: '10px',
}

const UserDataCard = (props) =>{
  if(props.dataObject.hasOwnProperty('user')) {
    return (
      <div style={rootDataCardStyle}>
        <img style={imageHeight} src={props.dataObject.user.avatars.default.https} alt="avatar"/>
        <div style={textBlock}>
          <div>
            <p style={pStyle}>Photographer: {props.dataObject.user.fullname}</p>
            <p style={pStyle}>Rating: {props.dataObject.rating}%</p>
          </div>
          <div>
            <img style={likeStyle} src={like} alt='like'/>
          </div>
        </div>
      </div>
    )
  }else{
    return <div style={{'display': 'none'}}/>
  }
}

export default UserDataCard

UserDataCard.propTypes={
  dataObject: PropTypes.object
}
