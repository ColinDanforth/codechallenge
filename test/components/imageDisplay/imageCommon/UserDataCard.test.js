import React from 'react';
import UserDataCard from '../../../../src/components/imageCommon/UserDataCard';
import renderer from 'react-test-renderer';

test('UserDataCard should display the appropriate strings', () => {
  const testDataObject={
    user: {
      avatars: {
        default: {
          https: 'http://google.ca'
        }
      },
      fullname: 'Test User',

    },
    rating: '99.99999995'
  }

  const component = renderer.create(
    <UserDataCard dataObject={testDataObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
