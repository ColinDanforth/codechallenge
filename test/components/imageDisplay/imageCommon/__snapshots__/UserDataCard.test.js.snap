// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDataCard should display the appropriate strings 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottom": "inset",
      "borderRight": "inset",
      "display": "flex",
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "flex-start",
      "maxWidth": "817px",
      "minWidth": "817px",
    }
  }
>
  <img
    alt="avatar"
    src="http://google.ca"
    style={
      Object {
        "height": "auto",
        "margin": "0.3%",
        "maxHeight": "100px",
        "width": "auto",
      }
    }
  />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "space-between",
        "marginLeft": "10px",
        "maxWidth": "700px",
        "minWidth": "700px",
        "width": "auto",
      }
    }
  >
    <div>
      <p
        style={
          Object {
            "height": "auto",
            "maxWidth": "400px",
            "minWidth": "400px",
            "width": "auto",
          }
        }
      >
        Photographer: 
        Test User
      </p>
      <p
        style={
          Object {
            "height": "auto",
            "maxWidth": "400px",
            "minWidth": "400px",
            "width": "auto",
          }
        }
      >
        Rating: 
        99.99999995
        %
      </p>
    </div>
    <div>
      <img
        alt="like"
        src="test-file-stub"
        style={
          Object {
            "height": "auto",
            "marginRight": "10px",
            "marginTop": "5px",
            "maxWidth": "40px",
            "width": "auto",
          }
        }
      />
    </div>
  </div>
</div>
`;
